java-checks/src/main/java/org/sonar/java/checks/UselessImportCheck.java

```diff
@@ -53,6 +53,7 @@ public class UselessImportCheck extends IssuableSubscriptionVisitor {
   private static final Pattern COMPILER_WARNING = Pattern.compile("The import ([$\\w]+(\\.[$\\w]+)*+) is never used");
   private static final Pattern NON_WORDS_CHARACTERS = Pattern.compile("\\W+");
   private static final Pattern JAVADOC_REFERENCE = Pattern.compile("\\{@link[^\\}]*\\}|(@see|@throws)[^\n]*\n");
+  private static final Pattern MARKDOWN_REFERENCE = Pattern.compile("\\[[^\\]]++\\]");
```

Comment:
```suggestion
  private static final Pattern MARKDOWN_REFERENCE = Pattern.compile("\\[[^\\[\\]]++\\]");
```
missing open bracket